docs/rfc/rfc-6.adoc

```diff
@@ -0,0 +1,219 @@
+delegated to opt-in maintainers and third parties to reduce the cost of
+operating a node.
+
+== In depth
+
+=== Transaction types
+
+On-chain transactions can be divided into three types:
+
+- Transactions that can only be submitted by group members, given that only the
+  group members have the knowledge about input parameters. For example, submit
+  DKG result transaction for ECDSA wallet or submit relay entry transaction for
+  the random beacon.
+  
+- Transactions that can be submitted by anyone, usually submitted often and
+  mostly belonging to a happy path. Never leading to slashing For example,
+  submit sweep proof or submit redemption proof transaction for tBTC v2. 
+
+- Transactions that can be submitted by anyone, submitted rarely or never, and
+  usually leading to slashing. For example, challenge DKG result for ECDSA
+  wallet or report fraud proof transaction for tBTC v2.
+
+For the first group of transactions, we will not enforce the order on-chain and
+we will expect off-chain clients to respect the submission order. If some client
+skipped their round, other clients can vote to mark the inactive client as
+ineligible for rewards. To disincentivize frontrunning, we will only reimburse
+ETH costs, and no additional reward will be allocated.
+
+The responsibility of executing the second group of transactions should be
+delegated to willing maintainers. Given that the preparation of the input data
+in some cases is complicated and may require access to the Bitcoin chain, the
+maintainer functionality should be built-in into the client and activated in the
+configuration. Eventually, we may decide to use networks like Gelato but given
+that the ECSDSA wallet could be slashed on the redemption proof timeout, such
+functionality should also be implemented in the client so that the client can
+protect the stake if required. Last but not least, Gelato Polywrap resolver
+allowing assembling more complicated transaction inputs is not yet available and
+we do not want to block the release on this functionality. Maintainers will be
+reimbursed in ETH and may receive an additional reward for performing their
+duties.
+
+Transactions belonging to the third type can be executed by anyone and all
+off-chain clients will be responsible for submitting them. No ETH will be
+reimbursed and T coming from slashed stakes will be the only optional reward.
+No submission order will be maintained and all clients will try to report
+misbehavior at the same time. It is more important to keep the network healthy
+and free from misbehavior than to optimize costs.
+
+==== Group member incentives
+
+===== Submit + approve DKG result
+
+No submission order will be enforced on-chain and the transaction submitter will
+have ETH cost reimbursed. To disincentivize front-running, no additional reward
+will be released by this transaction. Off-chain clients should have a handshake
+agreement to follow the submission order based on
```

Comment:
Is a handshake agreement like a verbal agreement or a gentleman's agreement? Or does it mean something else?

Reply:
What is the difference between a verbal agreement and a gentleman's agreement? If they are the same then yes, that's what I meant here. I'll clarify.

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/115/commits/35c11545a30a2bb2a9a4eef2df8cab7ccd15a124